### PR TITLE
feat: Add `num_epoch_to_store` to `config.json` - with test

### PR DESCRIPTION
**Pull request by pmnoxx**
Same as https://github.com/near/nearcore/pull/5348, but with extra test.

### CODE REVIEW

**Review by nikurt — APPROVED** [review]

**Comment by bowenwang1996** [comment]
@pmnoxx why not just modify the original PR to include the test?

File: integration-tests/tests/client/process_blocks.rs [hunk 1]
```diff
@@ -1558,6 +1575,34 @@ fn test_gc_after_state_sync() {
...
+        ))
+        .build();
+    for i in 1..epoch_length * (num_epochs_to_keep_store_data - 1) + 2 {
+        let block = env.clients[0].produce_block(i).unwrap().unwrap();
+        env.process_block(0, block.clone(), Provenance::PRODUCED);
+        env.process_block(1, block, Provenance::NONE);
+    }
+    // Should crash, but doesn't?
+    for i in 4..epoch_length * (num_epochs_to_keep_store_data - 1) + 2 {
+        env.clients[0].chain.get_block_by_height(i).unwrap();
```
Comment (mzhangmzz) [inline integration-tests/tests/client/process_blocks.rs:1602]: Does this test pass? 

Should you check the first few blocks are gc'ed?

Comment (pmnoxx) [reply integration-tests/tests/client/process_blocks.rs:1602]: @mzhangmzz This passes the test, but
Works:
``` (num_epochs_to_keep_store_data - 1) + 2```
Doesn't pass:
``` (num_epochs_to_keep_store_data) + 2```

You are right, I should also add a negative check in the test case.

Comment (pmnoxx) [reply integration-tests/tests/client/process_blocks.rs:1602]: @mzhangmzz I updated the test, to check whenever blocks get GC'ed

**Comment by jc-near** [comment]
Reopening this PR as [CP-25](https://nearinc.atlassian.net/jira/software/projects/CP/boards/17?selectedIssue=CP-25) requires it

**Comment by mm-near** [comment]
@jc-near - let's adopt this PR so that it uses the same 'GCconfig' that is added by #6592

**Comment by jc-near** [comment]
Meaning, once that PR is merged, we change this PR to use the same `GCconfig` struct and parametrize the variable there?

**Comment by jc-near** [comment]
@mm-near  ^^

**Comment by mina86** [comment]
> Meaning, once that PR is merged, we change this PR to use the same `GCconfig` struct and parametrize the variable there?

Yep. The idea now is to add gc_num_epochs_to_store to GCConfig struct. 

With the amount of conflicts this PR has it might be easier to just start anew.

**Comment by mm-near** [comment]
@jc-near - yes, it should use the GCconfig struct. And its up to you @jc-near whether you want to adapt this PR or start a new one.

**Review by mm-near — CHANGES_REQUESTED** [review]
.

**Comment by mina86** [comment]
Obsolete by 535f4d793

**Pull request closed without merging**